datasync/downloader/downloader_test.go

```diff
@@ -46,6 +49,48 @@ func init() {
 	fsHeaderContCheck = 500 * time.Millisecond
 }
 
+var (
+	lock   sync.Mutex
+	setter *govSetter
+)
+
+// govSetter sets governance items for testing purpose
+type govSetter struct {
+	numTesting          uint32
+	origStakingInterval uint64
+	origStakingManager  *reward.StakingManager
+}
+
+// setTestGovernance sets staking manager with memory db and staking update interval to 4.
+func setTestGovernance() {
+	lock.Lock()
+	defer lock.Unlock()
+	if setter == nil {
+		setter = &govSetter{
+			numTesting:          0,
+			origStakingInterval: params.StakingUpdateInterval(),
+			origStakingManager:  reward.GetStakingManager(),
+		}
+
+		reward.SetTestStakingManagerWithDB(database.NewMemoryDBManager())
+		params.SetStakingUpdateInterval(4)
+	}
+	setter.numTesting += 1
```

Comment:
I think we can use a waitgroup instead of just using a counter.
https://gobyexample.com/waitgroups

Reply:
Is this necessary? We don't need to wait for all goroutines to be terminated, but only to set governance before testing. I think also this is good to go..